msal4j-sdk/src/test/java/com/microsoft/aad/msal4j/HelperAndUtilityTests.java

```diff
@@ -187,4 +190,99 @@ void JsonHelper_createIdTokenFromEncodedTokenString_InvalidJsonInToken() {
 
         assertEquals(AuthenticationErrorCode.INVALID_JSON, exception.errorCode());
     }
+
+    @Test
+    void JwtHelper_buildJwt_UsesPSS256WhenSupported() throws Exception {
```

Comment:
The way it's implemented in MSAL .NET is: 

- for AAD, use SHA2 and PSS
- for ADFS, use SHA1 and PKCS1

But this is because MSAL.NET supports 
|The way it's implemented in MSAL .NET is: 

- for AAD, use SHA2 and PSS
- for ADFS, use SHA1 and PKCS1

I see that the way to specify a certificate in MSAL Java is similar. So why not have some higher level tests, i.e. start from the public API and assert what gets put on the wire?